forgot_password/handlers.py

```diff
@@ -131,69 +204,202 @@ def reset_password(user_id, code, new_password):
             userutil.set_new_password(c, user.id, new_password)
             logger.info('Successfully reset password for user.')
 
+            user_record = userutil.get_user_record(c, user.id)
+
+            # send welcome email
+            if welcome_email_settings.enable:
```

Comment:
Sending welcome email after the user has reset the password does not seem a desired behavior.